mithril-common/src/cardano_transaction_parser.rs

```diff
@@ -119,43 +116,54 @@ impl CardanoTransactionParser {
 
     /// Read blocks from immutable file
     fn read_blocks_from_immutable_file(immutable_file: &ImmutableFile) -> StdResult<Vec<Block>> {
-        let cbor = fs::read(&immutable_file.path).with_context(|| {
-            format!(
-                "CardanoTransactionParser could not read from immutable file: {:?}",
+        let cardano_blocks_reader =
+            CardanoTransactionParser::cardano_blocks_reader(immutable_file)?;
+
+        let mut blocks = Vec::new();
+        for block in cardano_blocks_reader {
+            let block = block.map_err(|e| {
+                anyhow!(e).context(format!(
+                    "Error while reading block in immutable file: '{:?}'",
+                    immutable_file.path
+                ))
+            })?;
```

Comment:
Why don't you use directly `with_context` ?
```suggestion
        for parsed_block in cardano_blocks_reader {
            let block = parsed_block.with_context(|| format!(
                    "Error while reading block in immutable file: '{:?}'",
                    immutable_file.path
                )
            )?;
```

Reply:
Done